File: Gemini/1. Algoritmo Básico Ordenamiento por Burbuja/java/src/Burbuja.java
import java.util.Arrays;
import java.util.Scanner;

public class Burbuja {
    public static void bubbleSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingrese una lista de números enteros separados por espacios: ");
        String entrada = scanner.nextLine();
        String[] numerosStr = entrada.split(" ");
        int[] numeros = new int[numerosStr.length];
        for (int i = 0; i < numerosStr.length; i++) {
            numeros[i] = Integer.parseInt(numerosStr[i]);
        }

        bubbleSort(numeros);
        System.out.println("Lista ordenada: " + Arrays.toString(numeros));
        scanner.close();
    }
}

File: Gemini/1. Algoritmo Básico Ordenamiento por Burbuja/C++/burbuja.cpp
#include <iostream>
#include <vector>
#include <sstream>
#include <algorithm>

std::vector<int> bubbleSort(std::vector<int> arr) {
    int n = arr.size();
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                std::swap(arr[j], arr[j + 1]);
            }
        }
    }
    return arr;
}

int main() {
    std::cout << "Ingrese una lista de números enteros separados por espacios: ";
    std::string entrada;
    std::getline(std::cin, entrada);
    std::stringstream ss(entrada);
    int numero;
    std::vector<int> numeros;
    while (ss >> numero) {
        numeros.push_back(numero);
    }

    std::vector<int> listaOrdenada = bubbleSort(numeros);
    std::cout << "Lista ordenada: ";
    for (int num : listaOrdenada) {
        std::cout << num << " ";
    }
    std::cout << std::endl;
    return 0;
}